### ISSUE TITLE
Add mechanism to turn off keyboard shortcuts (WCAG Level A requirement)

### ISSUE THREAD

**Issue by knowler** [npmx-dev/npmx.dev#1537]
npmx.dev implements single key shortcuts for various actions:

- globally:
	- <kbd>/</kbd> focuses search
	- <kbd>,</kbd> opens the settings
	- <kbd>?</kbd> highlights the keyboard hints
	- <kbd>c</kbd> opens compare
- in search
	- <kbd>↑</kbd> moves to the previous search result
	- <kbd>↓</kbd> moves to the next search result
	- <kbd>Enter</kbd> goes to the result (i.e. I believe the first)
- in the package view:
	- <kbd>c</kbd> opens compare for the package
	- <kbd>d</kbd> opens the generated API docs for the package
	- <kbd>.</kbd> opens the code for the package

[WCAG 2.2 Success Criterion 2.1.4: Character Key Shortcuts (Level A)](https://www.w3.org/TR/WCAG22/#character-key-shortcuts) states:

> If a [keyboard shortcut](https://www.w3.org/TR/WCAG22/#dfn-keyboard-shortcuts) is implemented in content using only letter (including upper- and lower-case letters), punctuation, number, or symbol characters, then at least one of the following is true:
> 
> <dl><dt>Turn off</dt><dd>
>
> A [mechanism](https://www.w3.org/TR/WCAG22/#dfn-mechanism) is available to turn the shortcut off;
> </dd>
> <dt>Remap</dt><dd>
>
> A mechanism is available to remap the shortcut to include one or more non-printable keyboard keys (e.g., Ctrl, Alt);
> </dd>
> <dt>Active only on focus</dt><dd>
>
> The keyboard shortcut for a [user interface component](https://www.w3.org/TR/WCAG22/#dfn-user-interface-components) is only active when that component has focus.
> </dd></dl>

Currently none of these conditions are satisfied and so **our current implementation fails this success criterion**.

## How to meet

If we want to keep shortcuts, **the easiest way to meet this requirement is to add a mechanism to turn them off**. This should also hide the keyboard hints on the page.

- [ ] Add a setting to disable keyboard shortcuts to the settings page.
- [ ] Add a note to the keyboard shortcuts modals that links directly to the setting.

Eventually, adding a way to remap the shortcuts could benefit users. For example, when I use NVDA, the screen reader’s default shortcuts override the website (e.g. arrow keys are used to navigate by item, <kbd>d</kbd> is used for navigating by landmark, <kbd>c</kbd> jumps to a combobox, etc.). Having a way to remap the keyboard shortcuts the website provides could benefit a user like this.

**Comment by alex-key** [comment]
I will try to implement this. If anyone is already working - please, leave a comment

**Comment by alex-key** [comment]
@knowler Do you think it worth adding 3 options in Settings:
- enable shortcuts
- enable shortcuts only with Alt pressed (probably this would require to display Alt + X on each button) <--- this is questionable
- disable shortcuts

**Comment by knowler** [comment]
@alex-key I would probably just do a toggle switch for now. I think the remapping case probably needs a bit of research.
